### PR TITLE
[WICKET-6935] several minor datatable improvements

### PR DESCRIPTION
**Pull request by reiern70**
Corresponding changes to 

https://github.com/apache/wicket/pull/481

But for mater. As agreed here we just change return type of factory method for label.

### CODE REVIEW

File: wicket-extensions/src/main/java/org/apache/wicket/extensions/markup/html/repeater/data/table/NavigationToolbar.java [hunk 1]
```diff
@@ -73,16 +68,16 @@ protected PagingNavigator newPagingNavigator(final String navigatorId,
...
 	 * @param navigatorId
 	 *            component id navigator label should be created with
 	 * @param table
-	 *            dataview used by datatable
+	 *            DataTable used by datatable
 	 * @return navigator label that will be used to navigate the data table
 	 * 
 	 */
-	protected WebComponent newNavigatorLabel(final String navigatorId, final DataTable<?, ?> table)
+	protected Component newNavigatorLabel(final String navigatorId, final DataTable<?, ?> table)
```
Comment (martin-g) [inline wicket-extensions/src/main/java/org/apache/wicket/extensions/markup/html/repeater/data/table/NavigationToolbar.java:80]: Please do the same for `org.apache.wicket.extensions.ajax.markup.html.AjaxEditableChoiceLabel#newLabel()`

**Review by martin-g — APPROVED** [review]

File: wicket-extensions/src/main/java/org/apache/wicket/extensions/markup/html/repeater/data/table/NavigationToolbar.java (hunk 1 above)
Comment (reiern70) [reply wicket-extensions/src/main/java/org/apache/wicket/extensions/markup/html/repeater/data/table/NavigationToolbar.java:80]: Done

**Comment by reiern70** [comment]
@martin-g many thanks for review. Landed in master
